bin/release

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env ruby
+
+require_relative "../lib/local_time/version"
+require "pathname"
+require "json"
+
+class Release
+  def initialize(version, dry: false)
+    @version = version
+    @dry = dry
+  end
+
+  def perform
+    write_version
+    build_gem
+
+    unless dry?
+      publish
+      cleanup
+    end
+
+    puts "== Done! =="
+  end
+
+  private
+    attr_reader :version, :dry
+
+    def write_version
+      puts "== Writing version to files =="
+
+      if Gem::Version.new(version) <= Gem::Version.new(LocalTime::VERSION)
+        abort "❌ New version (#{version}) must be greater than #{LocalTime::VERSION}"
+      end
+
+      write_version_to_rb
+      write_version_to_js
+    end
+
+    def write_version_to_rb
+      Pathname.new("./lib/local_time/version.rb").write <<~RUBY
+        module LocalTime
+          VERSION = "#{version}"
+        end
+      RUBY
+    end
+
+    def write_version_to_js
+      pathname = Pathname.new("package.json")
+      data = JSON.parse(pathname.read)
+      data["version"] = version
+      pathname.write(JSON.pretty_generate(data) + "\n")
+    end
+
+    def build_gem
+      puts "== Building gem =="
+
+      system "gem build local_time.gemspec"
+    end
+
+    def dry?
+      dry
+    end
+
+    def publish
+      puts "== Publishing releases =="
+
+      publish_git_release
+      publish_rubygems_release
+      publish_npm_release
+    end
+
+    def publish_git_release
+      abort("❌ Failed to commit") unless \
+        system("git add lib/local_time/version.rb package.json") &&
+        system("git commit -m 'Bump version for #{version}'") &&
+        system("git push") &&
+        system("git tag v#{version}") &&
+        system("git push --tags")
+
+      puts "✅ Successfully published git release"
+    end
+
+    def publish_rubygems_release
+      system "gem push 'local_time-#{version}.gem' --host https://rubygems.org"
```

Comment:
To release gems you can just do [`rake release`](https://guides.rubygems.org/releasing-rubygems/). It should take care of the proper naming for you.

Reply:
TIL!

Took a look at [what `rake release` does](https://github.com/rubygems/rubygems/blob/master/bundler/lib/bundler/gem_helper.rb#L65-L69C10) and it seems quite convenient for regular gems. It builds the gem, creates a git tag, performs the source control release, and pushes to rubygems in one go.

But since we're also writing the new version name in two files, bundling, committing that, and deploying to npm – and any one of those steps could fail, I think I'd rather have things spelled out as they are currently. I did grab some good ideas from `rake release` though: https://github.com/basecamp/local_time/pull/138/commits/6ec5f20e9661458558daad2119900113e878d762